agent/engine/daemonmanager/daemon_manager_linux.go

```diff
@@ -0,0 +1,166 @@
+	imageName := dm.managedDaemon.GetLoadedDaemonImageRef()
+	containerRunning := apicontainerstatus.ContainerRunning
+	dockerHostConfig := dockercontainer.HostConfig{
+		NetworkMode: apitask.HostNetworkMode,
+		// the default value of 0 for MaximumRetryCount means retry indefinitely
+		RestartPolicy: dockercontainer.RestartPolicy{
+			Name:              "on-failure",
+			MaximumRetryCount: 0,
+		},
+	}
+	if !dm.managedDaemon.IsValidManagedDaemon() {
+		return nil, fmt.Errorf("%s is an invalid managed daemon", dm.managedDaemon.GetImageName())
+	}
+	for _, mount := range dm.managedDaemon.GetMountPoints() {
+		err := mkdirAllAndChown(mount.SourceVolumeHostPath, 0700, daemonUID, os.Getegid())
+		if err != nil {
+			return nil, err
+		}
+		dockerHostConfig.Binds = append(dockerHostConfig.Binds,
+			fmt.Sprintf("%s:%s", mount.SourceVolumeHostPath, mount.ContainerPath))
+	}
+	rawHostConfig, err := json.Marshal(&dockerHostConfig)
+	if err != nil {
+		return nil, err
+	}
+	healthConfig := dm.managedDaemon.GetDockerHealthConfig()
+	rawHealthConfig, err := json.Marshal(&healthConfig)
+	if err != nil {
+		return nil, err
+	}
+	// The raw host config needs to be created this way - if we marshal the entire config object
+	// directly, and the object only contains healthcheck, all other fields will be written as empty/nil
+	// in the result string. This will override the configurations that comes with the container image
+	// (CMD for example)
+	rawConfig := fmt.Sprintf("{\"Healthcheck\":%s}", string(rawHealthConfig))
+
+	daemonTask := &apitask.Task{
+		Arn:                 fmt.Sprintf("arn:::::/%s-%s", dm.managedDaemon.GetImageName(), uuid.NewUUID()),
+		DesiredStatusUnsafe: apitaskstatus.TaskRunning,
+		Containers: []*apicontainer.Container{{
+			Name:                      dm.managedDaemon.GetImageName(),
+			Image:                     imageName,
+			ContainerArn:              fmt.Sprintf("arn:::::/instance-%s", dm.managedDaemon.GetImageName()),
+			Type:                      apicontainer.ContainerManagedDaemon,
+			TransitionDependenciesMap: make(map[apicontainerstatus.ContainerStatus]apicontainer.TransitionDependencySet),
+			Essential:                 true,
+			SteadyStateStatusUnsafe:   &containerRunning,
+			DockerConfig: apicontainer.DockerConfig{
+				Config:     aws.String(rawConfig),
+				HostConfig: aws.String(string(rawHostConfig)),
+			},
+			HealthCheckType: "DOCKER",
+		}},
+		LaunchType:         "EC2",
+		NetworkMode:        apitask.HostNetworkMode,
+		ResourcesMapUnsafe: make(map[string][]taskresource.TaskResource),
+		IsInternal:         true,
+	}
+	// add managed daemon environment to daemon task container
+	daemonTask.Containers[0].MergeEnvironmentVariables(dm.managedDaemon.GetEnvironment())
+	return daemonTask, nil
+}
+
+// LoadImage loads the daemon's latest image
+func (dm *daemonManager) LoadImage(ctx context.Context, _ *config.Config, dockerClient dockerapi.DockerClient) (*types.ImageInspect, error) {
```

Comment:
is _ *config.Config used in a later PR and you just have a placeholder for now?

Reply:
this will be used in a future pr, yes.  (I copied from the service connect manager)